### PR TITLE
Get a list of unique module's name

### LINKED ISSUE #34028: Multiple domains inside modules translations will slower the theme translations
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Once you have a theme (not Classic, as it's the core one) having somes modules views inside it that contains multiple string as the third part of the domain, the module will be inside the listing of the modules to translate each time.

It could (it does) slow the translations page.

### Expected behavior

Have an unique list, to faster the whole thing.

### Steps to reproduce

1. Install the child theme (or any other themes, as you want) like this : 
[child_classic.zip](https://github.com/PrestaShop/PrestaShop/files/12697983/child_classic.zip)
2. Go to translations page
3. Check the Front Office translations, with the child theme added
4. You will not be able to see without modifying a method to see the results, actually.
5. Ask a dev to confirm you that point, thanks.

### PrestaShop version(s) where the bug happened

8.1.2

### PHP version(s) where the bug happened

n/d

### If your bug is related to a module, specify its name and its version

n/d

### Your company or customer's name goes here (if applicable).

Wepika

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Get a list of unique module's name to avoid a slower foreach on theme translations.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See issue
| UI Tests          | n/d
| Fixed issue or discussion?     | Fixes #34028
| Sponsor company   | @Wepika

### CODE REVIEW

File: src/Core/Translation/Storage/Provider/ThemeCatalogueLayersProvider.php [hunk 1]
```diff
@@ -247,6 +247,7 @@ private function getModulesTranslations(string $locale): MessageCatalogue
     {
         $moduleCatalogue = new MessageCatalogue($locale);
         $modules = $this->getModulesFromTranslations($locale);
```
Comment (boherm) [inline src/Core/Translation/Storage/Provider/ThemeCatalogueLayersProvider.php]: Maybe it's better to do the `array_unique` at the end of `getModulesFromTranslations` function.
What do you think about it?

**Review by boherm — COMMENTED** [review]
Thanks for your contribution @PrestaEdit!
Just a little comment to discuss together, but I globally agree with this. 🙂

**Review by matthieu-rolland — COMMENTED** [review]
once @boherm 's feedback is applied it's lgtm for me 👍

**Review by boherm — APPROVED** [review]
lgtm now, thanks @PrestaEdit!

**Review by M0rgan01 — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Waiting for PM to confirm the expected behavior (see [comment](https://github.com/PrestaShop/PrestaShop/issues/34028#issuecomment-1739628999)).

**Review by LouisAUTHIE — APPROVED** [review]

**Comment by PrestaEdit** [comment]
Any update on this ?

It's 25 days without move.

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
thanks @PrestaEdit 

I also validated the PR, and the translations work as expected.

**Pull request merged**